ruleset: Add snat_port support for MASQUERADE

HSXX opened the pull request:
In a NAT444 dynamic IP environment where the ISP's BRAS uses Full-Cone NAT, limiting the MASQUERADE port range on OpenWrt can help prevent the number of sessions on the BRAS side from exceeding the limit. However, currently, if snat_port is configured in OpenWrt's NAT rules, the MASQUERADE rule fails to generate and produces an error. This modification ensures that MASQUERADE rules are correctly generated even when snat_port is specified.

HSXX committed:
ruleset: Add snat_port support for MASQUERADE

Currently, MASQUERADE in firewall4 does not support setting snat_port and directly reports an error without creating rules. This modification enables MASQUERADE to support setting snat_port.

Signed-off-by: HSXX <513733973@qq.com>

brada4 commented:
The code side is perfect. 
Please commit --amend ; push --force at least re-wording title (even it says everything there is) and adding S-o-b: which was likely wiped by githubs robots
See here for more info, your commit message goes to git history.
https://openwrt.org/submitting-patches#submission_guidelines

HSXX commented:
> The code side is perfect. Please commit --amend ; push --force at least re-wording title (even it says everything there is) and adding S-o-b: which was likely wiped by githubs robots See here for more info, your commit message goes to git history. https://openwrt.org/submitting-patches#submission_guidelines

Done. Fixed commit message and added S-o-b.

brada4 commented:
@jow- ?

Neustradamus commented:
It replaces:
- https://github.com/openwrt/firewall4/pull/78
